fv3core/tests/main/test_dycore_call.py

```diff
@@ -0,0 +1,162 @@
+    for (
+        obj
+    ) in dynamical_core._lagrangian_to_eulerian_obj._mapn_tracer._list_of_remap_objects:
+        original_attributes[obj] = obj._lagrangian_contributions
+        obj._lagrangian_contributions = do_nothing  # type: ignore
+    try:
+        yield
+    finally:
+        for obj, original in original_attributes.items():
+            obj._lagrangian_contributions = original
+
+
+def setup_dycore() -> Tuple[fv3core.DynamicalCore, List[Any]]:
+    backend = "numpy"
+    stencil_config = fv3core.StencilConfig(
+        backend=backend,
+        rebuild=False,
+        validate_args=True,
+    )
+    mpi_comm = NullComm(rank=0, total_ranks=6, fill_value=0.0)
+    namelist = Namelist.from_f90nml(f90nml.read(NAMELIST_FILENAME))
+    partitioner = pace.util.CubedSpherePartitioner(
+        pace.util.TilePartitioner(namelist.layout)
+    )
+    communicator = pace.util.CubedSphereCommunicator(mpi_comm, partitioner)
+    sizer = pace.util.SubtileGridSizer.from_tile_params(
+        nx_tile=namelist.npx - 1,
+        ny_tile=namelist.npy - 1,
+        nz=namelist.npz,
+        n_halo=3,
+        extra_dim_lengths={},
+        layout=namelist.layout,
+        tile_partitioner=partitioner.tile,
+        tile_rank=mpi_comm.rank,
+    )
+    grid_indexing = fv3core.GridIndexing.from_sizer_and_communicator(
+        sizer=sizer, cube=communicator
+    )
+    metric_terms = MetricTerms.from_tile_sizing(
+        npx=namelist.npx,
+        npy=namelist.npy,
+        npz=namelist.npz,
+        communicator=communicator,
+        backend=backend,
+    )
+
+    # create an initial state from the Jablonowski & Williamson Baroclinic
+    # test case perturbation. JRMS2006
+    state = baroclinic_init.init_baroclinic_state(
+        metric_terms,
+        adiabatic=namelist.adiabatic,
+        hydrostatic=namelist.hydrostatic,
+        moist_phys=namelist.moist_phys,
+        comm=communicator,
+    )
+    stencil_factory = fv3core.StencilFactory(
+        config=stencil_config,
+        grid_indexing=grid_indexing,
+    )
+
+    dycore = fv3core.DynamicalCore(
+        comm=communicator,
+        grid_data=GridData.new_from_metric_terms(metric_terms),
+        stencil_factory=stencil_factory,
+        damping_coefficients=DampingCoefficients.new_from_metric_terms(metric_terms),
+        config=namelist.dynamical_core,
+        phis=state.phis_quantity,
+    )
+    do_adiabatic_init = False
+    # TODO compute from namelist
+    bdt = 225.0
+
+    args = [
+        state,
+        namelist.consv_te,
+        do_adiabatic_init,
+        bdt,
+        namelist.n_split,
+    ]
+    return dycore, args
+
+
+def test_call_does_not_access_global_state():
+    dycore, args = setup_dycore()
+
+    def error_func(*args, **kwargs):
+        raise AssertionError("call not allowed")
+
+    mock_grid = unittest.mock.MagicMock()
+    with unittest.mock.patch("fv3core.utils.global_config.get_backend", new=error_func):
```

Comment:
Is the next step to eliminate `global_config` all together? Just realized that I could do that in physics today

Reply:
yes I have the same question -- if we remove the existence of this method, we shouldn't need to test for it, it should be removed. There's the possibility somebody will accidentally bring it back in merging this code with their branch, but we may forget to delete this when we are past the stage of people doing merges. I think that removing global state has been well communicated in the team, so I would think an accidental reverting of code would be caught by the merger, or by the reviewer.

Reply:
I believe `global_config.get_backend` is still used in our runfiles and in our test code. When we remove those functions, we need to be careful to re-run everything.

This test will make sure that even with the code there, it doesn't get called in the course of running the dycore.

So this is the next step, but it's not the current step (for this PR).